src/test/java/com/auth0/client/mgmt/JobsEntityTest.java

```diff
@@ -211,6 +213,55 @@ public void shouldThrowOnSendUserVerificationEmailWithNullUserId() {
         api.jobs().sendVerificationEmail(null, null);
     }
 
+    @Test
+    public void shouldSendUserVerificationEmailWithIdentity() throws Exception {
+        EmailVerificationIdentity identity = new EmailVerificationIdentity("google-oauth2", "1234");
+        Request<Job> request = api.jobs().sendVerificationEmail("google-oauth2|1234", "AaiyAPdpYdesoKnqjj8HJqRn4T5titww", identity);
+        assertThat(request, is(notNullValue()));
+
+        server.jsonResponse(MGMT_JOB_POST_VERIFICATION_EMAIL, 200);
+        Job response = request.execute();
+        RecordedRequest recordedRequest = server.takeRequest();
+
+        assertThat(recordedRequest, hasMethodAndPath("POST", "/api/v2/jobs/verification-email"));
+        assertThat(recordedRequest, hasHeader("Content-Type", "application/json"));
+        assertThat(recordedRequest, hasHeader("Authorization", "Bearer apiToken"));
+
+        Map<String, Object> body = bodyFromRequest(recordedRequest);
+        assertThat(body.size(), is(3));
+        assertThat(body, hasEntry("user_id", "google-oauth2|1234"));
+        assertThat(body, hasEntry("client_id", "AaiyAPdpYdesoKnqjj8HJqRn4T5titww"));
+
+        Map<String, String> identityMap = new HashMap<>();
+        identityMap.put("provider", "google-oauth2");
+        identityMap.put("user_id", "1234");
+        assertThat(body, hasEntry("identity", identityMap));
+
+        assertThat(response, is(notNullValue()));
+    }
+
+    @Test
+    public void shouldThrowOnSendUserVerificationEmailWithNullIdentityProvider() {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("'identity provider' cannot be null!");
+        api.jobs().sendVerificationEmail("google-oauth2|1234", null, new EmailVerificationIdentity(null, "user-id"));
+    }
+
+    @Test
+    public void shouldThrowOnSendUserVerificationEmailWithNullIdentityUserId() {
+        exception.expect(IllegalArgumentException.class);
+        // don't be too specific in testing the order we null-check provider and user-id
+        exception.expectMessage("'identity ");
```

Comment:
looking at the code, the line that should trigger this failure is 
```
Asserts.assertNotNull(emailVerificationIdentity.getUserId(), "identity user id");
```
I'd expect the message check to be more specific. You mention in your comment you don't want to, what is the reason to ignore that?

Reply:
Good question. Because the assert will throw a RuntimeException, the order we check the params determines which error message you receive. It's a question of how tightly coupled we want the test to be to the implementation details (which param we check first) when both are null. If we think that the order of the checks defines the contract of the API and should never change, we should verify that in the tests. I tend to think that order is not part of the API (they both throw the same type of exception), so didn't want to couple the test to that implementation detail.

Reply:
Ah, actually this check should be in the other test, `shouldThrowOnSendUserVerificationEmailWithNullIdentityProviderAndUserId` 😄. This one should indeed check the specific message. 

Reasoning still stands (willing to discuss further), just on the other test ;)